src/analyses/base.ml

```diff
@@ -1096,6 +1031,9 @@ struct
      * not include the flag. *)
     let update_one_addr (x, offs) (nst, fl, dep): store =
       let cil_offset = Offs.to_cil_offset offs in
+      let t = match t_override with
+        | Some t -> t
+        | None -> Cil.typeOf (Lval(Var x, cil_offset)) in
```

Comment:
Does this not cause us to have #119 again? (Until that's fixed by conditionally doing things for non-heap/heap variables.)

Reply:
Yes, it does.
This PR should probably be only merged once I have a working patch addressing #119.

Reply:
Should work now.